assets/js/blocks/product-query/inspector-controls.tsx

```diff
@@ -139,6 +139,7 @@ export const WooInheritToggleControl = (
 				setQueryAttribute( props, {
 					...props.defaultWooQueryParams,
 					inherit,
+					__woocommerceInherit: inherit,
```

Comment:
AFAIK, we should use `inherit` instead of `__woocommerceInherit`. The `__woocommerceInherit` is added only when `isCustomInheritGlobalQueryImplementationEnabled` is true, but we've decided not to implement the custom inherit query from template. Therefore, this flag will always be false, as seen [here](https://github.com/woocommerce/woocommerce-blocks/blob/adc72972bf76b0e2dee7ecf6a9ab25c2f7085639/assets/js/blocks/product-query/utils.tsx#L63). You can also find related discussion on Slack here: p1681812585287949-slack-C02UBB1EPEF.

In short, we shouldn’t use `__woocommerceInherit`. Additionally, I believe we can remove the `isCustomInheritGlobalQueryImplementationEnabled` flag and related code since we decided not to implement the "custom inherit query from template." I understand it might be confusing because there is still a partial implementation for the "custom inherit query from template," which is disabled by using the `isCustomInheritGlobalQueryImplementationEnabled` flag.

Does that make sense?

CC: @gigitux

Reply:
> In short, we shouldn’t use __woocommerceInherit. Additionally, I believe we can remove the isCustomInheritGlobalQueryImplementationEnabled flag and related code since we decided not to implement the "custom inherit query from template." I understand it might be confusing because there is still a partial implementation for the "custom inherit query from template," which is disabled by using the isCustomInheritGlobalQueryImplementationEnabled flag.

Since the logic is added on the Products block (that will be deprecated) I agree with you, we can remove that logic!

Reply:
> In [0d8fc18](https://github.com/woocommerce/woocommerce-blocks/pull/10303/commits/0d8fc1833cfe8f60015be5570f04346badab7828), I fixed the issue you mentioned in the review. I didn't handle well the case adding the block into archive template in the template editor.

Not sure why but it's still showing 9 products in Products beta block 🤔 

> I use the __woocommerceInherit attribute just for detecting if the current rest query is inherited from template. I used it in the rest_product_query filter [here](https://github.com/woocommerce/woocommerce-blocks/pull/10303/files#diff-efb947e345af5e501dfaa2980b055b3b31c308b31e8db23049b6b48fecee33d8R197) (as a custom [rest query argument](https://github.com/WordPress/gutenberg/blob/81b58d30a8685872fd5ac361bb75760d50318e1d/packages/block-library/src/post-template/edit.js#L190)). The reason for that is I can't find a better way, I'm open to any alternative that we can detect the 'inherit state' in the rest query. I admit this is a workaround than a solution, but given that the Products (Beta) block will be deprecated, this is acceptable IMO.

Got it. In that case, maybe we should consider naming this something other than __woocommerceInherit because its original intention was for use in the "custom inherit from query" implementation. Since this fix is not directly related to "custom inherit from query," using __woocommerceInherit for this fix might cause confusion. What do you think?

> About removing isCustomInheritGlobalQueryImplementationEnabled, let's do it in a separated PR for better history.

Definitely, Also this doesn't fall into scope of this PR.

Reply:
> Since this fix is not directly related to "custom inherit from query," using __woocommerceInherit for this fix might cause confusion. What do you think?

@imanish003 Because we're removing that "custom inherit feature", and the Products (Beta) block will be deprecated eventually, I don't think that's a big issue. But by the way, any name will work, I will try to find another one to clear any potential confusion.

> Not sure why but it's still showing 9 products in Products beta block 🤔 Are you able to replicate this?

Let's pair on this.

Reply:
>  But by the way, any name will work, I will try to find another one to clear any potential confusion.

Thank you so much for your understanding. Since we're unsure who will be working on removing the "custom inherit query" feature, it's better to avoid any potential confusion if we can. 🙌🏻